app/src/main/java/com/globant/calculatoronboardingkotlin/mvp/model/CalculatorModel.kt

```diff
@@ -1,17 +1,141 @@
 package com.globant.calculatoronboardingkotlin.mvp.model
 
 import com.globant.calculatoronboardingkotlin.mvp.contracts.CalculatorContracts
+import com.globant.calculatoronboardingkotlin.utils.Constants
+import com.globant.calculatoronboardingkotlin.utils.Constants.Companion.DIVIDE
+import com.globant.calculatoronboardingkotlin.utils.Constants.Companion.DIVIDING_BY_ZERO
+import com.globant.calculatoronboardingkotlin.utils.Constants.Companion.DOT
 import com.globant.calculatoronboardingkotlin.utils.Constants.Companion.EMPTY_STRING
+import com.globant.calculatoronboardingkotlin.utils.Constants.Companion.INPUTS_ARE_FULL
+import com.globant.calculatoronboardingkotlin.utils.Constants.Companion.MINUS
+import com.globant.calculatoronboardingkotlin.utils.Constants.Companion.MULTIPLY
+import com.globant.calculatoronboardingkotlin.utils.Constants.Companion.NUMBER_ZERO
+import com.globant.calculatoronboardingkotlin.utils.Constants.Companion.OPERATION_WITH_NO_NUMBER
+import com.globant.calculatoronboardingkotlin.utils.Constants.Companion.PLUS
+import com.globant.calculatoronboardingkotlin.utils.Constants.Companion.TOO_MANY_DOTS
+import com.globant.calculatoronboardingkotlin.utils.Constants.Companion.TOO_MANY_OPERATORS
+import com.globant.calculatoronboardingkotlin.utils.Constants.Companion.ZERO_DOUBLE
 
 class CalculatorModel : CalculatorContracts.Model {
 
     override var firstNumber: String = EMPTY_STRING
     override var secondNumber: String = EMPTY_STRING
     override var operator: String = EMPTY_STRING
+    var result: String = EMPTY_STRING
 
     override fun clear() {
         firstNumber = EMPTY_STRING
         secondNumber = EMPTY_STRING
         operator = EMPTY_STRING
+        result = EMPTY_STRING
+    }
+
+    override fun calculateResult(): String {
+        result = when (operator) {
+            PLUS -> firstNumber.toDouble() + secondNumber.toDouble()
+            MINUS -> firstNumber.toDouble() - secondNumber.toDouble()
+            MULTIPLY -> firstNumber.toDouble() * secondNumber.toDouble()
+            DIVIDE -> if (secondNumber.toDouble() == ZERO_DOUBLE) {
+                return DIVIDING_BY_ZERO
+            } else {
+                firstNumber.toDouble() / secondNumber.toDouble()
+            }
+            else -> EMPTY_STRING
+        }.toString()
+        firstNumber = result
+        operator = EMPTY_STRING
+        secondNumber = EMPTY_STRING
+        return result
+    }
+
+    override fun appendNumber(number: String): String {
+        var value: String = EMPTY_STRING
+        if ((result == firstNumber) && (operator.isEmpty())) {
+            clear()
+        }
+
+        if (operator == EMPTY_STRING) {
```

Comment:
can you use operator.isEmpty() like you are using in other places

Reply:
done, thanks